### ISSUE TITLE
Java's OnnxSequence::getValue leaking memory

### ISSUE THREAD

**Issue by vobornij** [microsoft/onnxruntime#6471]
Hi, I have a model which returns 
 - (tensor of) labels (string) and
 - **(sequence of) probabilities (map of string to float).**
When **extracting probabilities** from the native object using `OnnxSequence::getValue`, some **off-heap memory is allocated** and never freed. Everything works fine if I only extract labels or if we use int labels instead of strings.

OutputInfo from the object:
```
{output_label=NodeInfo(name=output_label,info=TensorInfo(javaType=STRING,onnxType=ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING,shape=[-1])), output_probability=NodeInfo(name=output_probability,info=SequenceInfo(length=UNKNOWN,type=MapInfo(size=UNKNOWN,keyType=STRING,valueType=FLOAT)))}
```
**Urgency**
We have a workaround that uses integer labels.

**System information**
- OS: Ubuntu 20.04
- JDK 11.0.4
- ONNX Runtime version: 1.6.0, CPU
- Python version: 3.8.5

**To Reproduce**
1) Create model in Python:
```
from skl2onnx import convert_sklearn
from skl2onnx.common.data_types import FloatTensorType
from sklearn.datasets import load_iris
from sklearn.ensemble import RandomForestClassifier
iris = load_iris()
X, y = iris.data, iris.target
y = list(map(str,y))
X = X[:, 0].reshape(-1, 1)
clr = RandomForestClassifier()
clr.fit(X, y)
initial_type = [('float_input', FloatTensorType([1, 1]))]
onx = convert_sklearn(clr, initial_types=initial_type,
                      target_opset=12)
with open("rf_iris.onnx", "wb") as f:
    f.write(onx.SerializeToString())
```

2) Run following Scala example code:
```
import java.nio.FloatBuffer
import java.nio.file.{Files, Paths}
import ai.onnxruntime.{OnnxTensor, OrtEnvironment, OrtSession}
import scala.jdk.CollectionConverters._

object Test {
  def loadModel(file: String, env: OrtEnvironment): OrtSession = {
    val bytes = Files.readAllBytes(Paths.get(file))
    env.createSession(bytes, new OrtSession.SessionOptions)
  }

  def main(args: Array[String]) = {
    val env = OrtEnvironment.getEnvironment
    val model =
      loadModel("/path/to/rf_iris.onnx", env)
    val shape = Array(1L, 1L)

    val buff = FloatBuffer.wrap(Array(1.0f))
    val inputTensor = OnnxTensor.createTensor(env, buff, shape)

    val input = Map("float_input" -> inputTensor)
    for (i <- 1 to 10 * 1000 * 1000) {
      val res = model.run(input.asJava)
      val probs =
        res
          .get("output_probability")
          .map(_.getValue) // <== this causes problems 

      res.close
    }

    inputTensor.close
    model.close
    env.close
  }
}
```

**Expected behavior**
Application memory should stay about constant after a short warmup time.

**Additional info**
I'm getting a warning about opset version 1 (we target 12 in skl2onnx). I'm not sure if it's relevant.
```
[W:onnxruntime:, model.cc:138 Model] ONNX Runtime only *guarantees* support for models stamped with opset version 7 or above for opset domain 'ai.onnx'. Please upgrade your model to opset 7 or higher. For now, this opset 1 model may run depending upon legacy support of some older opset version operators
```

**Comment by Craigacp** [comment]
I'll take a look.

**Comment by Craigacp** [comment]
Ah, looks like its missing `api->ReleaseValue(element);` [here]( https://github.com/microsoft/onnxruntime/blob/master/java/src/main/native/ai_onnxruntime_OnnxSequence.c#L30) and [here](https://github.com/microsoft/onnxruntime/blob/master/java/src/main/native/ai_onnxruntime_OnnxSequence.c#L82).

**Comment by vobornij** [comment]
Thanks, in my example it's about 375 bytes per call.

**Comment by Craigacp** [comment]
How big are the returned Strings?

**Comment by vobornij** [comment]
The strings are "0", "1", "2".

**Comment by Craigacp** [comment]
Ok. Well that seems like a lot, but I don't know how big the OrtValue is internally as I only interact with it through the C API. I looked through the allocation path and those are the only frees missing, so I'll put in a PR with the fix shortly.

**Comment by snnn** [comment]
Fixed merged.

**Comment by vobornij** [comment]
Could you please estimate when this will be released? Our workaround breaks some monitoring dashboards which would take some effort to fix.

**Comment by Craigacp** [comment]
Looks like `rel-1.7.0` has been branched from `master`, so it's probably going to be relatively soon, but maybe someone from MS's PM team could comment? cc @faxu

**Comment by Craigacp** [comment]
See #6786
